src/python/pants/backend/shell/target_types.py

```diff
@@ -30,14 +30,19 @@
 from pants.util.enums import match
 
 
-class ShellSources(Sources):
+class ShellSourceField(Sources):
```

Comment:
I'm not in love with this name, particularly wondering if it should be `Source` vs. `Sources`. 

I think the superclass will probably always be `Sources`, which we cannot rename to `Source` because there are non-file based atomic targets like `go_package` which need >1 source file. So, weird for subclass to be `Source` and parent to be `Sources`.

But `ShellSourcesField` also feels wrong - we want to express in the type system that Shell operates with files as its atoms.

Reply:
Also, I'm thinking about how to model subclasses like this expressing with the type system that they are file-based, while still having `Sources` as a superclass. Something like either a class-decorator or subclassing `SourceField` (which itself subclasses `Sources`). It would rename the field alias to be `source` and expect a single `str`, rather than `list[str]`.

Reply:
I think that's better, even it the extra characters are annoying.

Also, I tried this weekend experimenting with subclasses of `Sources` abusing the class to be `source: str` instead of `sources: list[str]`, and it was chaotic. Pretty sure it violates Liskov Substitution and is a bad idea. If we want a `source` vs. `sources` field, then I think we need two distinct classes and for all our supporting code (`graph.py` etc) to work with both. That's not worth doing for now, so I think we stick with `sources` and set `expected_num_files = 1`.

Reply:
I think `ShellSourcesField` is fine - the fact that we operate on single sources at a time is somewhat an implementation choice. So I don't think we want to encode that in the name. Imagine a language where the theoretical atom is a single file, but in practice we choose to act on a few files together for whatever reason.  There is no reason that code consuming a `ShellSourcesField` should care that it happens to represent a single source? That decision has been made upstream.

Reply:
Sgtm. Thanks Benjy!